flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/UpsertKafkaTableITCase.java

```diff
@@ -385,6 +384,7 @@ private void wordFreqToUpsertKafka(String wordCountTable) throws Exception {
 	}
 
 	private void writeChangelogToUpsertKafkaWithMetadata(String userTable) throws Exception {
+		env.setParallelism(1);
```

Comment:
I would suggest to move the parallelism setting to `testTemporalJoin()` before calling `writeChangelogToUpsertKafkaWithMetadata()` and add a note. Otherwise, other contributors may don't understand this and change it in the future.